### ISSUE TITLE
Spam

### ISSUE THREAD

**Issue by CassieH2018** [stcr/subscribe-to-comments-reloaded#642]
I'm getting a bunch of spam sign ups a day for my comments and this started approximately a month ago.  Everyday I have a bunch of undelivered email notices from these spammy sign ups, and that have hit real inboxes too (for random companies and such). I read a forum on here that suggested I remove the option to subscribe to comments without leaving a comment which I did two days ago, but the problem persists.  I've also tried setting up the authenticator option but it didn't work.  Fresh out of ideas.

**Comment by BobaWebDev** [comment]
Hi,

If the "subscribe without commenting" is disabled then it shouldn't be possible to have new signups unless a comment is left.

Can you send over the URL to the site?

**Comment by CassieH2018** [comment]
www.cookitrealgood.com

**Comment by BobaWebDev** [comment]
Hi @CassieH2018 

Are you getting spam subscription signups or spam requests for a subscription management link?

The form for the subscription management link is still active at https://www.cookitrealgood.com/comment-subscriptions/

You can disable that as well, it's in WP admin > StCR > Management Page. You'll see "Allow Request Management Link".

If you want to keep it you can set up some anti spam measures which the plugin has. They are in WP admin > StCR > Options.

You'll see the "Challenge question" options in there. There's also "Google reCAPTCHA" if you prefer to go that way.

**Comment by CassieH2018** [comment]
I removed all that you pointed out and added the question whats 1+2 = 3 and I woke up this morning to 50 undeliverable emails saying they couldn't subscribe people to me posts.  this is so odd!

**Comment by BobaWebDev** [comment]
Hi @CassieH2018 

I went to the site to give it a try (to leave a comment and subscribe) but the plugin seems to be deactivated at the moment.

**Comment by CassieH2018** [comment]
sorry it's back up now! Thanks

**Comment by BobaWebDev** [comment]
Is the issue that they did not receive the confirmation emails or that the confirmation link in the email didn't work?

I tried subscribing but did not receive a confirmation email yet. The way to test if the email functionality is called is to install [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/) plugin.

After installing (and activating) that plugin will log all email requests and show if there were any errors. 

Can you please install that plugin and let me know when you do.

**Comment by CassieH2018** [comment]
Ok I activated that.  You've actually identified a completely new issue which is the user isn't getting email notification - I tested this morning too and didn't work.  

The original issue was I was getting hundreds of spammy "subscribes" everyday without leaving a comment which came on all of a sudden in the past month after using the plugin for 4 years.  I have removed the option to subscribe without commenting and am waiting to see if they stop, as I still had some failed to deliver emails hit my inbox this morning saying that an email address that subscribed wasn't deliverable.

**Comment by BobaWebDev** [comment]
Hi @CassieH2018 

Do the comment notifications show up in the list (from WP Mail Logging) and are there any errors shown?

**Comment by CassieH2018** [comment]
it's saying in that plugin that emails are sending and there aren't any errors - but when I did a test I still didn't get an email notifying me of a reply (even though i selected i wanted to be notified).  checked spam folders and everything.

I'm also still getting spammy subscribes to my posts without comment, which i removed the functionality for so I don't understand how this is happening

**Comment by BobaWebDev** [comment]
Hi @CassieH2018 

**Spam**

Is it still happening now as well? 

**Emails not received**

If WP Mail Log shows the entry and does not show an error for it then the most likely issue is that for some reason the hosting provider stops the email from being sent.

Do other emails from that WordPress install work fine? Like a contact form for example.

If those emails get through but the ones from our plugin don't, it's definitely blocked by the server. In which case you need to access the control panel (for the hosting) and check the "mail error logs". In case you're not sure where to find that, contact the support for the hosting, they'll help out with that.
